#!/usr/bin/env python3
PACKAGE = "val_moveit_planner_executor"

from dynamic_reconfigure.parameter_generator_catkin import *

# initialize parameter generator to define parameters
gen = ParameterGenerator()

# add parameters to generator
gen.add("moveit_im_update_topic", str_t,    0,
        "Topic for listening to MoveIt Interactive Marker; must publish messages of type InteractiveMarkerUpdate",
        "/rviz_moveit_motion_planning_display/robot_interaction_interactive_marker_topic/update")
gen.add("pos_tolerance",          double_t, 0,
        "Position tolerance for motion planning goals (not waypoint trajectories)",
        0.01, 0.0, 5.0)
gen.add("ang_tolerance",          double_t, 0,
        "Angular tolerance for motion planning goals (not waypoint trajectories)",
        0.01, 0.0, 5.0)
gen.add("max_cartesian_step",     double_t, 0,
        "Maximum distance (in Cartesian space) between consecutive points in the planned trajectory",
        0.1, 0.0, 0.5)
gen.add("max_configuration_step", double_t, 0,
        "Maximum distance (in configuration space) between consecutive points in the planned trajectory",
        5.0, 0.0, 5.0)

# generate necessary files
exit(gen.generate(PACKAGE, "val_moveit_planner_executor", "IMPlanningRequestParams"))
